LeoBirds/Engine/Vehicle.ixx

```diff
@@ -0,0 +1,63 @@
+﻿export module Engine.Vehicle;
+
+import <memory>;
+import <stdexcept>;
+import <SFML/System/Vector2.hpp>;
+
+import FLib.DrawableImage;
+import BoatCharacteristics;
+import GunCharacteristics;
+
+namespace birds
+{
+    export class Vehicle
+    {
+    public:
+        //Constructor of gun
+        explicit Vehicle(float fireRate, int health, int damage);
+        //Constructor of boat
+        explicit Vehicle(sf::Vector2f position, int health, int damage, float speed);
+        ~Vehicle() = default;
+
+
+        //fire() is declared to trigger gun fire from the main gun
+        void fire();
+        //takeDamage() is declared to trigger the fact of loosing life from the player or boats
+        void takeDamage();
+
+    private:
+        sf::Vector2f m_position;
+        std::unique_ptr<BoatCharacteristics> m_boatCharacteristics;
+        std::unique_ptr<GunCharacteristics> m_gunCharacteristics;
+    };
+}
+
+module: private;
+
+namespace birds
+{
+    Vehicle::Vehicle(float fireRate, int health, int damage)
+    {
+        m_gunCharacteristics->Health = health;
+        m_gunCharacteristics->Damage = damage;
+        m_gunCharacteristics->FireRate = fireRate;
+    }
+
+    Vehicle::Vehicle(sf::Vector2f position, int health, int damage, float speed)
+    {
+        m_boatCharacteristics->Health = health;
+        m_boatCharacteristics->Damage = damage;
+        m_boatCharacteristics->Speed = speed;
+        m_position = position;
+    }
+
+    void Vehicle::fire()
+    {
+        throw std::logic_error("Not implemented");
+    }
+
+    void Vehicle::takeDamage()
+    {
+        m_boatCharacteristics->Health - m_gunCharacteristics->Damage;
```

Comment:
unused statement
should it be m_boatCharacteristics->Health -= m_gunCharacteristics->Damage